### PR TITLE
Added Files OAuth for File and Directory Rename

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.Shares/src/autorest.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/21921735e92f74e027266506d019135ccf0d4f8b/specification/storage/data-plane/Microsoft.FileStorage/preview/2022-11-02/file.json
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/043006cf67b3858f9d77216e95a7366297ef7edf/specification/storage/data-plane/Microsoft.FileStorage/preview/2022-11-02/file.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/autorest.md:7]: TODO update this when https://github.com/Azure/azure-rest-api-specs/pull/22420 merged

File: sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public ShareClientTestFixtureAttribute()
                 StorageVersionExtensions.MaxVersion)
         {
             RecordingServiceVersion = StorageVersionExtensions.MaxVersion;
-            LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion, };
+            LiveServiceVersions = new object[] { StorageVersionExtensions.MaxVersion, };
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs:32]: Revert this.

Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs:32]: Is this just a name change? or is there a difference between latest and max version

**Review by jaschrep-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs (hunk 2 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs:32]: woops. I didn't see your other comment

**Review by amnguye — APPROVED** [review]

**Pull request merged**